svm-decode/src/lib.rs

```diff
@@ -169,40 +229,57 @@ pub fn decode_instructions(
     let data_arrays: Vec<Arc<dyn Array>> = decoded_params_vec
         .iter()
         .enumerate()
-        .map(|(i, v)| to_arrow(&signature.params[i].param_type, v.clone()).unwrap())
+        .map(|(i, v)| {
+            to_arrow(&signature.params[i].param_type, v.clone())
+                .context("unable to convert instruction value to a arrow format value")
+                .unwrap()
+        })
         .collect::<Vec<_>>();
 
     let data_fields = signature
         .params
         .iter()
-        .map(|p| Field::new(p.name.clone(), to_arrow_dtype(&p.param_type).unwrap(), true))
+        .map(|p| {
+            Field::new(
+                p.name.clone(),
+                to_arrow_dtype(&p.param_type)
+                    .context("unable to convert instruction param type to arrow dtype")
+                    .unwrap(),
+                true,
+            )
+        })
         .collect::<Vec<_>>();
 
     let acc_names_len = signature.accounts_names.len();
+    let mut accounts_arrays = Vec::new();
+    let mut acc_fields = Vec::new();
 
-    let mut accounts: Vec<Arc<dyn Array>> = accounts
-        .iter()
-        .map(|arr| {
+    for i in 0..acc_names_len {
+        let arr = accounts
+            .get(i)
+            .context(format!("Account a{} not found during decoding", i))?;
+        if let Some(arr) = arr {
             let owned_array = arr.slice(0, arr.len());
-            owned_array as Arc<dyn Array>
-        })
-        .collect();
-
-    let mut acc_fields = Vec::new();
-    if acc_names_len < 10 {
-        let _ = accounts.split_off(acc_names_len);
-        for i in 0..acc_names_len {
-            let field = Field::new(signature.accounts_names[i].clone(), DataType::Binary, true);
-            acc_fields.push(field);
+            accounts_arrays.push(Arc::new(owned_array) as Arc<dyn Array>);
+        } else {
+            return Err(anyhow::anyhow!(
+                "Account a{} is Null, but required by the signature",
+                i
+            ));
         }
-    } else {
-        for i in 0..10 {
+        if signature.accounts_names[i].is_empty() {
```

Comment:
might make more sense to name this field account_names

Reply:
??

Reply:
just removing the 's' from accounts? This goes to python =/

Reply:
Yeah not worth it 👍